core/src/tile/tileWorker.cpp

```diff
@@ -67,6 +67,9 @@ void TileWorker::run() {
                     if (a->tile->isVisible() != b->tile->isVisible()) {
                         return a->tile->isVisible();
                     }
+                    if (a->tile->sourceGeneration() != b->tile->sourceGeneration()) {
+                        return a->tile->sourceGeneration() < b->tile->sourceGeneration();
```

Comment:
This should be when both `a` and `b` tiles belong to the same `source`, else `priority` should be checked.

Can you also update the comment above (line 64) to reflect this change in logic? Thanks